lib/segment/src/index/field_index/numeric_index/mod.rs

```diff
@@ -325,6 +340,43 @@ impl<T: Encodable + Numericable + Default> NumericIndexInner<T> {
     pub fn values_is_empty(&self, idx: PointOffsetType) -> bool {
         self.values_count(idx) == 0
     }
+
+    pub fn values_by_key(&self, key: &T) -> Box<dyn Iterator<Item = PointOffsetType> + '_> {
+        let start = Bound::Included(Point::new(*key, PointOffsetType::MIN));
+        let end = Bound::Included(Point::new(*key, PointOffsetType::MAX));
+        match &self {
+            NumericIndexInner::Mutable(mutable) => Box::new(mutable.values_range(start, end)),
+            NumericIndexInner::Immutable(immutable) => Box::new(immutable.values_range(start, end)),
+        }
+    }
+
+    /// Tries to estimate the amount of points for a given key.
+    pub fn estimate_points(&self, key: &T) -> usize {
+        let start = Bound::Included(Point::new(*key, PointOffsetType::MIN));
+        let end = Bound::Included(Point::new(*key, PointOffsetType::MAX));
+
+        match &self {
+            NumericIndexInner::Mutable(mutable) => {
+                let mut iter = mutable.map().range((start, end));
+                let first = iter.next();
+                let last = iter.next_back();
+
+                match (first, last) {
+                    (Some(_), None) => 1,
+                    (Some(start), Some(end)) => (start.idx..end.idx).len(),
+                    (None, _) => 0,
+                }
+            }
+            NumericIndexInner::Immutable(immutable) => {
+                let start_index = immutable.map.find_start_index(start);
+                let end_index = immutable.map.find_end_index(start_index, end);
+                let range_size = end_index - start_index;
+                let avg_values_per_point =
+                    self.total_unique_values_count() as f32 / self.get_points_count() as f32;
+                (range_size as f32 / avg_values_per_point).round() as usize
```

Comment:
When `range_size` is `1` and `avg_values` is `3`, the result will be `0`. I propose to avoid zero returning if `range_size` is not zero